event-hubs.api.md

```diff
@@ -0,0 +1,303 @@
+}
+
+// @public
+export interface LastEnqueuedEventInfo {
+    enqueuedTime?: Date;
+    offset?: string;
+    retrievalTime?: Date;
+    sequenceNumber?: number;
+}
+
+export { MessagingError }
+
+// @public
+export interface OwnershipManager {
+    claimOwnership(partitionOwnership: PartitionOwnership[]): Promise<PartitionOwnership[]>;
+    listOwnership(fullyQualifiedNamespace: string, eventHubName: string, consumerGroupName: string): Promise<PartitionOwnership[]>;
+}
+
+// @public
+export interface PartitionCheckpointer {
+    updateCheckpoint(eventData: ReceivedEventData): Promise<void>;
+    updateCheckpoint(sequenceNumber: number, offset: number): Promise<void>;
+    // @internal (undocumented)
+    updateCheckpoint(eventDataOrSequenceNumber: ReceivedEventData | number, offset?: number): Promise<void>;
+}
+
+// @public
+export interface PartitionContext {
+    consumerGroupName: string;
+    eventHubName: string;
+    fullyQualifiedNamespace: string;
+    partitionId: string;
+}
+
+// @public
+export type PartitionManager = CheckpointManager & OwnershipManager;
+
+// @public
+export interface PartitionOwnership extends PartitionContext {
+    eTag?: string;
+    lastModifiedTimeInMS?: number;
+    offset?: number;
+    ownerId: string;
+    ownerLevel: number;
+    sequenceNumber?: number;
+}
+
+// @public
+export interface PartitionProperties {
+    beginningSequenceNumber: number;
+    eventHubName: string;
+    lastEnqueuedOffset: number;
+    lastEnqueuedSequenceNumber: number;
+    lastEnqueuedTimeUtc: Date;
+    partitionId: string;
+}
+
+// @public
+export type ProcessCloseHandler = (reason: CloseReason, context: PartitionContext & PartitionCheckpointer) => Promise<void>;
+
+// @public
+export type ProcessErrorHandler = (error: Error, context: PartitionContext) => Promise<void>;
+
+// @public
+export type ProcessEvents = (receivedEvent: ReceivedEventData, context: PartitionContext & PartitionCheckpointer) => Promise<void>;
+
+// @public
+export type ProcessInitializeHandler = (context: PartitionContext) => Promise<void>;
+
+// @public
+export interface ReceivedEventData {
+    body: any;
+    enqueuedTimeUtc: Date;
+    offset: number;
+    partitionKey: string | null;
+    properties?: {
+        [key: string]: any;
+    };
+    sequenceNumber: number;
+    systemProperties?: {
+        [key: string]: any;
+    };
+}
+
+export { RetryOptions }
+
+// @public
+export interface SendBatchOptions {
+    abortSignal?: AbortSignalLike;
+    parentSpan?: Span | SpanContext;
+}
+
+// @public
+export interface Subscription {
+    close(): Promise<void>;
+    isRunning(): boolean;
+}
+
+// @public
+export interface SubscriptionEventHandlers {
+    processClose?: ProcessCloseHandler;
+    processError?: ProcessErrorHandler;
+    processEvent: ProcessEvents;
+    processInitialize?: ProcessInitializeHandler;
+}
+
+// @public
+export interface SubscriptionOptions extends SubscriptionEventHandlers {
+    defaultEventPosition?: EventPosition;
```

Comment:
This should not be part of the `SubscriptionOptions`
When receiving from single partition, the position should be a mandatory argument for `subscribe()`

Reply:
Quick check - we didn't talk about the mandatory argument portion.

As it is now the user can set the event position as part of the processInitialize handler - this is in any overload since all overloads support passing handlers. Do we need a separate EventPosition argument?

Reply:
Fair point.

Technically, we dont need a separate EventPosition argument if we can make use the `processInitialize` handler, but usage wise it feels odd to do so.

For someone who want to receive from a single partition, `processInitialize` and `processClose` doesnt really make sense, does it?

Reply:
They still seem as sensible as they are with multiple partitions - it lets you encode your entire processing lifetime in a single set of handlers.

Reply:
(I've left both in place for this API review - there's not a clear answer)

Reply:
Latest after offline discussions is to remove this. @richardpark-msft Please update the file.